clients/devtools-entry.js

```diff
@@ -29,18 +29,24 @@ function getDefaultConfigForCategories(categoryIDs) {
 
 /**
  * @param {RawProtocol.Port} port
+ * @returns {RawProtocol}
+ */
+function setUpWorkerConnection(port) {
+  return new RawProtocol(port);
+}
+
+/**
  * @param {string} url
  * @param {LH.Flags} flags Lighthouse flags.
  * @param {Array<string>} categoryIDs Name values of categories to include.
+ * @param {RawProtocol} connection
  * @return {Promise<LH.RunnerResult|void>}
  */
-function runLighthouseInWorker(port, url, flags, categoryIDs) {
+function runLighthouse(url, flags, categoryIDs, connection) {
```

Comment:
at this point, it's hard to pretend this function is encapsulating anything from `AuditsService.js` :) so how about we just get rid of it?

It would be very helpful for e.g. testing in #9924 if `runLighthouse` was just the normal `lighthouse` entry point. I didn't want to make any changes to the interface in https://github.com/GoogleChrome/lighthouse/pull/9943#discussion_r343914283 since it would require changes on the DevTools side, but if we're making changes there anyways...

The only thing left after the changes here is if a config was passed in instead of `categoryIDs`, and that could just be another helper function like `setUpWorkerConnection` (just put `getDefaultConfigForCategories` on `self`).

Reply:
so you're suggesting we just 

```js
self.runLighthouse = lighthouse;
self.getDefaultConfigForCategories = catIds => getDefaultConfigForCategories(catIds);
```

ya?

ill prep a PR into this PR

Reply:
> ya?

ya!

Reply:
https://github.com/chruxin/lighthouse/pull/1

Reply:
Thanks! Merged your commit